test/sql/test_deprecations.py

```diff
@@ -17,19 +23,31 @@
 from sqlalchemy import util
 from sqlalchemy.engine import default
 from sqlalchemy.schema import DDL
+from sqlalchemy.sql import functions
 from sqlalchemy.sql import util as sql_util
+from sqlalchemy.sql.functions import GenericFunction
+from sqlalchemy.sql.sqltypes import NullType
 from sqlalchemy.testing import assert_raises
 from sqlalchemy.testing import assert_raises_message
 from sqlalchemy.testing import AssertsCompiledSQL
 from sqlalchemy.testing import engines
 from sqlalchemy.testing import eq_
 from sqlalchemy.testing import fixtures
 from sqlalchemy.testing import mock
+from sqlalchemy.testing.assertions import expect_warnings
 
 
 class DeprecationWarningsTest(fixtures.TestBase):
     __backend__ = True
 
+    def setup_method(self):
```

Comment:
**mike bayer** (zzzeek) wrote:

hmmm is setup_method / teardown_method a pytest thing?  I think the test suite uses setup() / teardown() across the board so we should keep consistent.

Reply:
Yes, this is a `pytest` feature to execute the setup and teardown methods for each test method, not only at the beginning and the end of the test class. I needed that because the content of the registry at the beginning of a given test (e.g. `test_deprecated::test_case_sensitive`) was dependant of the number of threads used for running the tests. This is not because of thread safety, it is just because the previous tests were updating the registry, so the tests were actually mutually dependant.

Reply:
that's what setup() and teardown() do.   setup() and teardown() are per test.  setup_class() and teardown_class() are per test class.  Within py.test this might be part of its "nose" compatibility, SQLAlchemy just dropped nose support in 1.3.    But in any case, any future changes to test suite idioms should be done all at once and I don't want to introduce any pytest-specific idioms (since we did that with Nose, and then suddenly Nose decided to go away, so never again with that).

Reply:
Ok, `setup()` and `teardown()` seems to work but not `tear_down()`. So if I just change `tear_down()` it seems ok.